build-logic/src/main/kotlin/io/sentry/gradle/AggregateJavadoc.kt

```diff
@@ -0,0 +1,36 @@
+package io.sentry.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileCollection
+import org.gradle.api.file.FileSystemOperations
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import javax.inject.Inject
+
+abstract class AggregateJavadoc @Inject constructor(
+    @get:Internal val fs: FileSystemOperations
+) : DefaultTask() {
+    @get:InputFiles
+    abstract val javadocFiles: Property<FileCollection>
+
+    @get:OutputDirectory
+    abstract val outputDir: DirectoryProperty
+
+    @TaskAction
+    fun aggregate() {
+        javadocFiles.get().forEach { file ->
+            fs.copy {
+                // Get the third to last part (project name) to use as the directory name for the output
+                val parts = file.path.split('/')
+                val projectName = parts[parts.size - 4]
```

Comment:
Should there be some safety around this in case the `parts.size` is < 4?

Reply:
This is a good point. To be honest, this logic is a bit hacky. The `ArtifactView` has this information as the component idenitifer, but I can’t have the ArtifactView as an input to the task since it isn’t serializeable. This was my hack around this.

Since these are fully qualified paths it is nearly impossible for them to be less than 4. If they are then we should fail and investigate. The bigger issue is if these are not the project names but I’m not sure how to check for that.